### DISCUSSION TITLE
Multiple DirectoryHarvesting and SourceDir placeholder

### DISCUSSION THREAD

**Discussion by SirDester** [wixtoolset/issues#8067] [Questions] [answered]
Hello, I started using WiX a few weeks ago so sorry If my question is silly and/or my assumptions are wrong.

First thing is: Is it mandatory to add a HarvestDirectory and a BindPath with the same folders for every harvesting?
Then suppose you have 2 (or more) folders to harvest. Looking at the documentation I have to add the following elements to the project file

```
<HarvestDirectory Include="FolderA">...</HarvestDirectory>
<BinPath Include="FolderA">
<HarvestDirectory Include="FolderB">...</HarvestDirectory>
<BinPath Include="FolderB">
```

Then suppose in FolderA you have a file named "mylibrary.dll" and in BolderB a file with the same name, but a different version (odd, but it can happen).

Both the harvesting generates two compiled files with the same File element 

```
<File .... Name="SourceDir\mylibrary.dll">
```

and, if I understand well, the SourceDir is a placeholder to all the BinPath declared in the project, and they are scanned sequentially to find the files to add.

So my question/doubt: how can I be sure that the right "mylibrary.dll" file is choosen from every folder if folders are scanned sequentially?

I saw that the <HarvestDirectory> element has the PreprocessorVariable attribute that replaces the SourceDir with a custom variable... so in case of multiple harvesting **is it mandatory to set different properties for every folders and set the relative PreprocessorVariable to remove the ambiguity** ?

I think documentation lacks some information about this (or I'm too young to understand all the process eheheh).

Thanks all
Maurizio

**Comment by robmen** [comment, marked as answer]
I think you understand the situation quite well. Heat-based harvesting is just a bit limited. In [S1:E52 of the Deployment Dojo I walk through how using heat is harder](https://robmensching.com/deployment-dojo/episodes/s1/e52/harvesting-the-hard-way-with-heat-in-wix-v4/) than using something like the [advanced harvesting tools from FireGiant that I covered in the next episode S1:E53](https://robmensching.com/deployment-dojo/episodes/s1/e53/harvesting-simplified-by-the-heatwave-build-tools-for-wix-v4/).

**Comment by SirDester** [comment]
Thanks for your fast response,
I'm using extensively your Dojo videos and I find them really interesting, well done and well explained in details, and a good way for a jump start in learning WiX fundamentals.

For the Harvesting I fully saw the Episode 52 and I started from that... but i needed to watch the following episode :)
I continue to study, and thanks for your work.

Maurizio.

**Reply by robmen** [reply]
You're doing great.

Also, good to hear that you enjoy the [Deployment Dojo show](https://robmensching.com/deployment-dojo/) and find it useful.
